clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -175,81 +80,71 @@ export class CopybookDownloadService {
     );
   }
 
-  public makeResolveCopybookHandler() {
-    return this.resolveCopybookHandler.bind(this);
-  }
-
-  public makeCopybookDownloadHandler() {
-    return (
-      cobolFileName: string,
-      copybookNames: CopybookName[],
-      _quietMode: boolean,
-    ) => {
-      return this.downloadCopybooks(cobolFileName, copybookNames);
-    };
+  public makeResolveCopybookUriHandler() {
+    return this.resolveCopybookURI.bind(this);
   }
 
-  public async resolveCopybookHandler(
-    documentUri: string,
+  async resolveCopybookURI(
+    documentURI: string,
     copybookName: string,
     dialectType: string,
   ): Promise<string | undefined> {
-    if (this.handleAsEndevorElement(documentUri)) {
-      const copybookUri = await this.e4eDownloader?.getE4ECopyBookLocation(
+    // is endevor -> download copybook from endevor and return local Uri
+    if (this.handleAsEndevorElement(documentURI)) {
+      const endevorResult = await this.e4eDownloader?.downloadCopybookE4E(
+        documentURI,
         copybookName,
-        documentUri,
+        dialectType,
       );
-      return copybookUri?.toString();
+
+      return endevorResult?.toString();
     }
+    // search processor groups
     const pgConfigs = await loadProcessorGroupCopybookPathsConfig(
-      { scopeUri: documentUri },
+      { scopeUri: documentURI },
       [],
       dialectType,
     );
+
     if (pgConfigs.length > 0) {
-      return (
-        await searchCopybookinProcessorGroups(
-          documentUri,
+      const processorGroupsResult =
+        await this.resolveCopybookUriInProcessorGroups(
           copybookName,
-          this.storagePath,
+          "profile",
```

Comment:
Shouldn't this call contain an actual default profile rather than the `"profile"` string?